src/database.h

```diff
@@ -161,11 +159,8 @@ class DBResult
 			}
 
 			T data;
-			try {
-				data = boost::lexical_cast<T>(row[it->second]);
-			} catch (boost::bad_lexical_cast&) {
-				data = 0;
-			}
+			std::stringstream ss(row[it->second]);
+			ss >> data;
```

Comment:
LGTM. The static istringstream object should be thread local to prevent race conditions though.